src/utils/DateUtils.ts

```diff
@@ -1,24 +1,125 @@
+   *
+   * **Examples**
+   *
+   * - If the date to render is equal to the current date, it will display 'Today':
+   *    Current date: march 8 2017, date to render: march 8 2017, result: 'Today'
+   *
+   * - If the date to render is equal to the current date plus or minus 1, it will display Yesterday or Tomorrow:
+   *    Current date: march 8 2017, date to render: march 7 2017, result: 'Yesterday'
+   *
+   * Default value is 'true'
+   */
   useTodayYesterdayAndTomorrow?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or using days of the week if the date it is part of the
+   * current week.
+   *
+   * **Example**
+   *
+   * -If the date to render has less than a week's difference with the current date, it will display the corresponding
+   * day:
+   *    Current date: monday march 8 2017, date to render: march 11 2017, result: 'Next Thursday'
+   *
+   * Default value is 'true'
+   */
   useWeekdayIfThisWeek?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or without the year.
+   *
+   * **Example**
+   *
+   * - If the date to render is the same year as the current one, it won't display the year value:
+   *    Current date: march 8 2017, date to render: 09/22/2017, result: 'September 22'
+   *
+   * Default value is 'true'
+   */
   omitYearIfCurrentOne?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or displayed using the complete format.
+   *
+   * **Example**
+   *
+   * - If the date to render is 08/04/2017:
+   *    The result will be Friday, August 04, 2017
+   *
+   * Default value is 'false'
+   */
   useLongDateFormat?: boolean;
+
+  /**
+   * Specifies whether the date should be dislayed normally or if it should also display the time if it is today.
+   *
+   * **Example**
+   *
+   * - If the date to render is equal to the current date, it will display the day and the time:
+   *    Current date: march 8th 2017, date to render: 03/08/2017 7:10, result: 'Today, 7:10 AM'
+   *
+   * Default value is 'true'
+   */
   includeTimeIfToday?: boolean;
+
+  /**
+   * Specifies whether the date should be dislayed normally or if it should also display the time if it is part of
+   * the current week.
+   *
+   * **Example**
+   *
+   * - If the date to render is equal to the current date, it will display the day and the time:
+   *    Current date: march 8th 2017, date to render: 03/10/2017 7:10, result: 'March 03, 7:10 AM'
+   *
+   * Default value is 'true'
+   */
   includeTimeIfThisWeek?: boolean;
+
+  /**
+   * Specifies whether the date should be dislayed normally or if it should also display the time.
+   *
+   * **Example**
+   *
+   * - If the option is 'true', the time will always be displayed:
+   *    Current date: march 8th 2017, date to render: 04/11/2015 7:10, result: '04/11/2015, 7:10 AM'
+   *
+   * Default value is 'false'
+   */
   alwaysIncludeTime?: boolean;
+
```

Comment:
What is the option precedence? I'm guessing this one would have precedence over any other option. Is that correct?

> Specifies a custom format to apply to the resulting string (e.g., `YYYY-MM-dd`).
>
> See the [Moment.js Format](http://momentjs.com/docs/#/displaying/format/) documentation.
>
> Default value is `undefined`.

Please make sure the documentation link I'm suggesting is actually relevant :)